nemo/src/io/parser.rs

```diff
@@ -1611,4 +1659,42 @@ mod test {
             )
         );
     }
+
+    #[test]
+    fn program_statement_order() {
+        assert_matches!(
+            parse_program(
+                r#"@output s .
+                   s(?s, ?p, ?o) :- t(?s, ?p, ?o) .
+                   @source t[3]: load-rdf("triples.nt") .
+                  "#
+            ),
+            Ok(_)
```

Comment:
We have this, not sure if an end-to-end test would provide any additional value.

Reply:
yes very much much optional